tests/integration/interface_common_test.py

```diff
@@ -34,3 +44,59 @@ def test_iface_description_removal(eth1_up):
     libnmstate.apply(desired_state)
     current_state = statelib.show_only(('eth1',))
     assert Interface.DESCRIPTION not in current_state[Interface.KEY][0]
+
+
+def test_take_over_virtual_interface_then_remove():
+    assert _add_ip_link_dummy(DUMMY_INTERFACE) == 0
+    dummy_desired_state = {
+        Interface.KEY: [
+            {
+                Interface.NAME: DUMMY_INTERFACE,
+                Interface.STATE: InterfaceState.UP,
+                Interface.IPV4: {InterfaceIPv4.ENABLED: False},
+                Interface.IPV6: {InterfaceIPv6.ENABLED: False},
+            }
+        ]
+    }
+    libnmstate.apply(dummy_desired_state)
```

Comment:
This and the teardown part needs to be placed in a context manager with a try/finally clause, otherwise we may be left with leftovers from the test.

Reply:
Done